ctapipe_io_lst/event_time.py

```diff
@@ -0,0 +1,293 @@
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import TelescopeParameter
+from traitlets import Enum, Int as _Int, Bool
+from astropy.time import Time
+from astropy.table import Table
+import numpy as np
+from collections import deque, defaultdict
+
+
+CENTRAL_MODULE = 132
```

Comment:
@morcuended : is it really worth to move away from module 82, used until now in lstchain for this purpose? I am afraid the values in the NightSummary files (giving the correspondence between DRS4 clock ticks & UCTS time) may be different for the two modules, and this would imply having to re-run the Night Summary for all nights. After all, there is nothing special about the central module which makes it more reliable.

Reply:
As I understood, lstchain did not even use module 82, but the module at position 82 in the file. Which could be any module?

Reply:
May be... @morcuended  can comment. If that is the case, is Isidro's Night Summary also doing the same?
Should we move in LSTOSA towards getting the numbers with ctapipe_io_lst from sub-run 0?

Reply:
Catching up with this issue. In principle from Isidro's check, there are only a few cases in which some modules do not have 10MHz counters well synchronized with the rest. In those cases, the timestamp could differ at most 100 ns.

As @maxnoe said, lstchain was wrongly using index 82 to identify the central module 132.

Regarding Night Summary, according to Isidro, he has always been identifying the central module 132 with index 82, and I even got the impression that for the timestamps he uses the mode of the values among all the modules (this is still to be confirmed).

I would say that we should adopt the same convention (only central module 132, mode or median among all of them, etc) for both Night Summary and lstchain as of when this `ctapipe_io_lst` stage1 starts working.

Reply:
Hi @morcuended, can you confirm that it was not module id=82, but just the random module which appeared at position 82? (both in lstchain and in Isidro's code).
How long would it take for Isidro to re-generate all the night summaries?

Reply:
> Hi @morcuended, can you confirm that it was not module id=82, but just the random module which appeared at position 82? (both in lstchain and in Isidro's code).

In lstchain, we were using the module corresponding to index 82, *which is not module 82*, i.e.:
```
event.lst.tel[telescope_id].evt.pps_counter[82]
```
Back then I thought this 82 index was always matching the central module (number 132), being the relationship between module ID and module index one to one and always the same. I checked the first subrun over more than 1000 runs from Nov 2019 to Dec 2020 and it turns out that index 82 corresponds always to module 133 for the first event:
```
event.lst.tel[telescope_id].svc.module_ids[82] = 133
```
Of course, this does not mean that this correspondence is always the same for all the events of all the runs.

Concerning Isidro's code, that is producing those initial timestamps, the central module was always used to get the timestamps until September. As of then, the initial timestamps written in the NightSummary files were obtained using the most repeated value among all the modules in order to avoid the situation in which the central module is not available. Indeed the values obtained in the two ways should be exactly the same in most of the cases.

In cases in which not all modules operate synchronously, the difference in the timestamps has never been seen larger than 200 ns according to Isidro.

All in all, there should be no need to reprocess the NightSummary files. The initial timestamps should be fine for the bulk of the runs regardless of the module you use to build the timestamps. Maybe we should think of the possibility of not relying only on one module to avoid the potential situation in which this particular module is not working.

Reply:
I did an initial check some weeks ago (cannot check right now because fefs is online) but I found clearly that the module order differs from run to run. I took a random run from each day if I remember correctly.